Case: Campaign Testing

A company wants to run a new campaign to increase the transactions’ amount of its product. Before running the actual campaign to the whole customers, the company wants to know the campaign’s effectiveness in their 30 days transactions’ amount.
The average of transactions’ amount before the campaign is `$100`.
The company ran a testing on 20 customers and gathered their ransactions’ amount in the next 30 days after the campaign. The data is stored in this table below.z
Run a t-test analysis on the transactions’ amount average (with alpha = 5%). 

In [1]:
# Import library
import scipy.stats as stats
import numpy as np
import pandas as pd

In [3]:
# Dataframe Transaction
df = pd.DataFrame({
    "transaction_amount": [100,150,50,100,130,120,100,110,75,65],
    "income": [150,120,50,100,100,140,90,150,50,90]
})
df

transaction_amount  income
0                 100     150
1                 150     120
2                  50      50
3                 100     100
4                 130     100
5                 120     140
6                 100      90
7                 110     150
8                  75      50
9                  65      90

In [8]:
# Mengambil data pada kolom income
data = np.array(list(df['income']))
data

array([150, 120,  50, 100, 100, 140,  90, 150,  50,  90])

### Pendefinisian Hipotesis (H0 dan H1):
- H0: Rata-rata penjualan setelah Training = `$100`
- H1: Rata-rata penjualan setelah Training > `$100`

### Melakukan T-Test

In [6]:
# perform one sample t-test
stats.ttest_1samp(a=data, popmean=100)

Ttest_1sampResult(statistic=0.345834250622541, pvalue=0.7374138103020644)

Hasil yang didapatkan 
- T-statistic = 0.345834250622541
- P-value = 0.7374138103020644

Jika menggunakan alpha = 5% = 0.05, maka P-value > alpha, artinya kita gagal menolak H0

In [7]:
# pvalue / 2 -> untuk dibandingkan dengan alpha dalam menentukan daerah critical region
0.7374138103020644/2

0.3687069051510322

- alpha = 0.05, dengan P-value/2 = 0.3687069051510322. 
- Karena P-value/2 > alpha, artinya kita gagal untuk menolak H0
- Conclusion: The Mean is equal to 100

If using T-statistic, we have to find the Critical Region first (One Tailed):

In [10]:
# If H1: mean > 100 (right side)
stats.t.ppf(1-0.05, 9)

1.8331129326536333

Hasil yang diperoleh:
- Critical Region: t > 1.8331129326536333
- T-statistic = 0.345834250622541
- Karena T-statistic tidak lebih besar dari critical region atau tidak berada dalam critical region, maka kita gagal untuk menolak H0
- Conclusion: The mean is equal to 100

### Kesimpulan: Rata-rata penjualan setelah pelatihan = $100. Artinya training tidak berhasil atau campaign yang dilakukan gagal untuk meningkatkan transactions amount dari produknya.